File: 3555-final-array-state-after-k-multiplication-operations-i/3555-final-array-state-after-k-multiplication-operations-i.java
class Solution {
    public int[] getFinalState(int[] nums, int k, int multiplier) {
        while(k > 0) {
            int mini=Integer.MAX_VALUE, idx=0;
            for(int i=0;i<nums.length;i++) {
                if(nums[i] < mini) {
                    mini=nums[i];
                    idx=i;
                }
            }

            nums[idx]*=multiplier;
            k--;
        }
        return nums;
    }
}

File: 3555-final-array-state-after-k-multiplication-operations-i/3555-final-array-state-after-k-multiplication-operations-i.py
class Solution(object):
    def getFinalState(self, nums, k, multiplier):
        """
        :type nums: List[int]
        :type k: int
        :type multiplier: int
        :rtype: List[int]
        """
        while(k>0):
            min=sys.maxsize
            idx=0
            for i in range(len(nums)):
                if(nums[i] < min):
                    min=nums[i]
                    idx=i
            nums[idx]*=multiplier
            k-=1
   
        return nums
                    
        